### PR TITLE
Introduce non-profile connections and UI stubs

### PR DESCRIPTION
**Pull request by rli**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    fun createConnection(profile: AuthProfile): ToolkitConnection
+
+    fun deleteConnection(connection: ToolkitConnection)
+    fun deleteConnection(connectionId: String)
+
+    fun getConnection(connectionId: String): ToolkitConnection?
+}
+
+interface ToolkitConnectionManager {
```
Comment (JadenSimon) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:51]: Needs a `logout` (or `signout`?) stub method. Just call `deleteConnection` on the active connection in the impl.

**Review by JadenSimon — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:57]: went with a message bus approach for now

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettingsMenus.kt [hunk 2]
```diff
@@ -200,3 +262,36 @@ class SettingsSelectorComboLabel(private val selectorLogic: SettingsSelectorLogi
         label.toolTipText = selectorLogic.tooltip()
     }
 }
+
+class CredsComboBoxActionGroup(private val project: Project) : ActionGroup("group", false) {
+    private val toolkitConnectionAction = ToolkitConnectionComboBoxAction(project)
+    private val profileRegionSelectorGroup: Array<AnAction> by lazy {
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettingsMenus.kt:268]: Should this be DumbAware Action?

**Review by manodnyab — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettingsMenus.kt (hunk 2 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettingsMenus.kt:268]: groups don't need to be dumbaware

**Pull request merged**